### PR TITLE
Lt 2077 route of goods

### PR DESCRIPTION
**Pull request by stephanieboyd**

### CODE REVIEW

File: applications/serializers/end_use_details.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from rest_framework import serializers
 
 from applications.models import BaseApplication
+from applications.serializers.serializer_helper import _validate_field
```
Comment (Mark-Spiers-kainos) [inline applications/serializers/end_use_details.py]: you are importing a method with an _

File: applications/serializers/open_application.py [hunk 2]
```diff
@@ -7,8 +7,10 @@
     GenericApplicationViewSerializer,
 )
 from applications.serializers.licence import LicenceViewSerializer
+from applications.serializers.serializer_helper import _validate_field
```
Comment (Mark-Spiers-kainos) [inline applications/serializers/open_application.py]: and again

File: applications/tests/tests_adding_route_of_goods.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    @parameterized.expand([[CaseTypeEnum.F680], [CaseTypeEnum.EXHIBITION], [CaseTypeEnum.GIFTING]])
+    def test_non_open_or_standard_applications_failure(self, case_type):
+        case = self.create_mod_clearance_application(self.organisation, case_type=case_type)
+        url = reverse("applications:route_of_goods", kwargs={"pk": case.id})
+        response = self.client.put(url, self.data, **self.exporter_headers)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
+            response.json()["errors"], ["This operation can only be used on applications of type: open, standard"]
```
Comment (Mark-Spiers-kainos) [inline applications/tests/tests_adding_route_of_goods.py]: Is this error not in strings?

File: applications/serializers/serializer_helper.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from rest_framework import serializers
+
+
+def _validate_field(data, field_name, error, required=False):
```
Comment (iichr) [inline applications/serializers/serializer_helper.py]: that'd come in handy now it's in helpers 👍

File: applications/serializers/standard_application.py [hunk 5]
```diff
@@ -114,3 +127,18 @@ def _update_have_you_been_informed_linked_fields(cls, instance, validated_data):
...
         else:
             instance.reference_number_on_information_form = None
+
+    def validate(self, data):
+        _validate_field(
+            data,
+            "is_shipped_waybill_or_lading",
+            strings.Applications.Generic.RouteOfGoods.IS_SHIPPED_AIR_WAY_BILL_OR_LADING,
+        )
+        if data.get("is_shipped_waybill_or_lading") == False:
```
Comment (iichr) [inline applications/serializers/standard_application.py]: `validate_field` returns the field value already, so could just assign it `is_shipped_waybill_or_lading = validate_field...` and reuse

File: applications/views/applications.py [hunk 6]
```diff
@@ -702,3 +713,48 @@ def post(self, request, application):
             return JsonResponse(data={"application": serializer.data}, status=status.HTTP_200_OK)
 
         return JsonResponse(data={"errors": serializer.errors}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class ApplicationRouteOfGoods(UpdateAPIView):
+    authentication_classes = (ExporterAuthentication,)
+
+    @authorised_users(ExporterUser)
+    @application_in_major_editable_state()
```
Comment (iichr) [inline applications/views/applications.py]: so this works with the tests right? might take advantage of that instead of the end use details approach :)

**Pull request merged**